### ISSUE TITLE
Bug: equal sign in path breaks lambda dynamic URL paths

### ISSUE THREAD

**Issue by baraksalomon** [awslabs/aws-lambda-powertools-python#1736]
### Expected Behaviour

When having an path like: /token/bXl0b2tlbg==
And a decorator like:
@app_rest.post("/token/<token>")
def post_token(token: str):
..

The resolve should find this function and token parameter should be populated with "bXl0b2tlbg=="

### Current Behaviour

When having an path like: /token/bXl0b2tlbg==
And a decorator like:
@app_rest.post("/token/<token>")
def post_token(token: str):
..

The resolve returns 404.

### Code snippet

```python
from aws_lambda_powertools.event_handler import APIGatewayRestResolver

app = APIGatewayRestResolver()

minimal_event = {
    "path": "/token/bXl0b2tlbg==",
    "httpMethod": "POST"
}


@app.post("/token/<token>")
def post_token(token: str):
    return {"token": token}


def lambda_handler(event, context):
    return app.resolve(event, context)


if __name__ == "__main__":
    print(lambda_handler(minimal_event, None))
```

### Possible Solution

Didn't test it but maybe the equal sign is missing from:
_SAFE_URI = "-._~()'!*:@,;"  # https://www.ietf.org/rfc/rfc3986.txt

https://github.com/awslabs/aws-lambda-powertools-python/blob/118f3e995f35808cc3e4edbc70c33e6769e2506e/aws_lambda_powertools/event_handler/api_gateway.py#L43

### Steps to Reproduce

Run the snipper code.
returns 404 instead of 200

### AWS Lambda Powertools for Python version

latest

### AWS Lambda function runtime

3.9

### Packaging format used

PyPi

**Comment by heitorlessa** [comment]
@baraksalomon thank you for flagging it - I've just tested an exact route and works (API GW decodes the `=`) but fails as a dynamic argument.

Creating a test and a fix shortly.

Thanks!

**Comment by heitorlessa** [comment]
Done - triple tested with an endpoint deployed with the fix just in case. Waiting CI to complete and will kick off a patch release to unblock you and prevent others from failures.

**Endpoint - GET**: https://p7bquief72.execute-api.eu-west-1.amazonaws.com/Prod/token/bXl0b2tlbg==

**Snippet**

```python
from aws_lambda_powertools.event_handler import APIGatewayRestResolver
from aws_lambda_powertools.utilities.typing import LambdaContext

app = APIGatewayRestResolver()

@app.get("/token/<token>")
def get_token(token: str):
    return {"received": f"{token}"}


def lambda_handler(event: dict, context: LambdaContext) -> dict:
    return app.resolve(event, context)

```

**Comment by baraksalomon** [comment]
Wow, that was super fast.
Thanks you!

**Comment by heitorlessa** [comment]
Fix available as 2.3.1 in PyPi - Compiling Lambda Layer and deploying the next version (15) across all regions.

**Comment by heitorlessa** [comment]
All complete now (PyPi+Layers+Docs) - let us know if this somehow doesn't fix it for you @baraksalomon.

Thank you for the great bug report one more time ;)
